addon/record-array.js

```diff
@@ -6,6 +6,11 @@ export default class extends RecordArray {
   // TODO: implement more of RecordArray but make this not an arrayproxy
 
   replace(idx, removeAmt, newModels) {
+    if (typeof this._replace === 'function') {
```

Comment:
instead of doing this kind of check let's split this

`RecordArray` - base class
`QueryArray < RecordArray` - used by `queryURL`
`ReferenceArray < RecordArray` - used for arrays of attribute references